src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -152,5 +182,23 @@ public static CompositeConfigurationSource CreateDefaultConfigurationSource()
 
             return configurationSource;
         }
+
+        internal bool IsIntegrationEnabled(string name)
+        {
+            return TraceEnabled &&
+                   Integrations[name].Enabled != false &&
+                   !DisabledIntegrationNames.Contains(name);
+        }
+
+        internal double? GetIntegrationAnalyticsSampleRate(string name, bool enabledWithGlobalSetting)
+        {
+            var integrationSettings = Integrations[name];
+            var analyticsEnabled = integrationSettings.AnalyticsEnabled ?? (enabledWithGlobalSetting && AnalyticsEnabled);
+            var sampleRate = integrationSettings.AnalyticsSampleRate ?? AnalyticsSampleRate;
+
+            return analyticsEnabled
+                       ? sampleRate
+                       : (double?)null;
```

Comment:
interesting syntax... I get it... just calling out it is interesting. (re: `(double?)null`)

Reply:
TL;DR: the code won't compile without the cast.

In the expression `a ? b : c`, `b` and `c` need to be of the same type or have an implicit conversion between them. In this case, we have a `double` and `null`. There is no implicit conversion between them, but both can be converted implicitly to `double?`. The compiler just needs help figuring this out. As long as one of the values is `double?` the other can be either `double` or `null`.

Casting `sampleRate` would have worked just as well:
```csharp
// cast sampleRate instead of null
return analyticsEnabled
           ? (double?)sampleRate
           : null;
```